lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -1079,64 +1079,81 @@ def test_source_is_empty(self):
     @ddt.data(
         (
             {
-                'desktop_webm': 'https://webm.com/dw.webm',
-                'hls': 'https://hls.com/hls.m3u8',
                 'youtube': 'v0TFmdO4ZP0',
-                'desktop_mp4': 'https://mp4.com/dm.mp4'
+                'hls': 'https://hls.com/hls.m3u8',
+                'desktop_mp4': 'https://mp4.com/dm.mp4',
+                'desktop_webm': 'https://webm.com/dw.webm',
             },
             ['https://www.youtube.com/watch?v=v0TFmdO4ZP0']
         ),
         (
             {
-                'desktop_webm': 'https://webm.com/dw.webm',
-                'hls': 'https://hls.com/hls.m3u8',
                 'youtube': None,
-                'desktop_mp4': 'https://mp4.com/dm.mp4'
+                'hls': 'https://hls.com/hls.m3u8',
+                'desktop_mp4': 'https://mp4.com/dm.mp4',
+                'desktop_webm': 'https://webm.com/dw.webm',
             },
-            ['https://hls.com/hls.m3u8']
+            ['https://www.youtube.com/watch?v=3_yD_cEKoCk']
         ),
         (
             {
-                'desktop_webm': 'https://webm.com/dw.webm',
-                'hls': None,
                 'youtube': None,
-                'desktop_mp4': 'https://mp4.com/dm.mp4'
+                'hls': None,
+                'desktop_mp4': None,
+                'desktop_webm': None,
             },
-            ['https://mp4.com/dm.mp4']
+            ['https://www.youtube.com/watch?v=3_yD_cEKoCk']
         ),
+    )
+    @ddt.unpack
+    @patch('xmodule.video_module.video_module.HLSPlaybackEnabledFlag.feature_enabled', Mock(return_value=True))
+    def test_val_encoding_in_context(self, val_video_encodings, video_url):
```

Comment:
What's external url in this case ?

Reply:
external url is `https://www.youtube.com/watch?v=3_yD_cEKoCk` in this case. It is the default external YT encoding with which video component is initialized.